library/src/main/java/org/apache/apex/malhar/lib/window/impl/KeyedWindowedOperatorImpl.java

```diff
@@ -135,6 +147,14 @@
   }
 
   @Override
+  public void endWindow()
+  {
+    super.endWindow();
+
+    updatedKeyStorage.endWindow();
```

Comment:
Instead of calling endWindow explicitly, it should be part of the components member variable